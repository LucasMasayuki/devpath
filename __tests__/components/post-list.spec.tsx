import React from 'react'
import renderer from 'react-test-renderer';
import PostList from '@/components/post-list';
import Author from '@/types/author'
import Post from '@/types/post'

describe('PostList', () => {
  it('should render PostList', () => {
    const author: Author = {
      name: 'test',
      picture: 'test',
    }

    const post: Post = {
      content: 'test',
      slug: '',
      title: 'test',
      coverImage: 'test.com',
      date: '1996-10-02',
      author: author,
      excerpt: 'test',
      ogImage: {
        url: 'test'
      },
    }

    const posts: Post[] = [
      post,
    ];

    const component = renderer.create(
      <PostList posts={posts}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})