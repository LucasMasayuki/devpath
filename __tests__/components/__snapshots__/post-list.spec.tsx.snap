// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList should render PostList 1`] = `
<div
  className="css-j7qwjs"
>
  <div
    className="css-15jqh4k"
  >
    <div
      className="css-1fo89uv"
    >
      <div
        style={
          Object {
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "left": 0,
            "margin": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <img
          alt="Cover Image for test"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-n16x53"
    >
      <div
        className="css-j7qwjs"
      >
        <div
          className="css-0"
        >
          <h3
            className="chakra-heading css-1dv4iqg"
          >
            <a
              className="chakra-link css-0"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              test
            </a>
          </h3>
        </div>
        <div
          className="css-17xejub"
        />
        <div
          className="css-0"
        >
          <div
            className="chakra-stack css-180nsro"
          >
            <span
              className="chakra-avatar css-1ruhx4w"
            >
              <div
                aria-label="test"
                className="chakra-avatar__initials css-0"
              >
                t
              </div>
            </span>
            <span
              className="css-g5hezf"
            >
              test
            </span>
          </div>
        </div>
        <div
          className="css-17xejub"
        />
        <div
          className="css-0"
        >
          <time
            dateTime="1996-10-02"
          >
            October	2, 1996
          </time>
           -
           
          <svg
            className="chakra-icon css-zmpx8u"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
              />
              <circle
                cx="12"
                cy="12"
                r="2"
              />
            </g>
          </svg>
          <small>
            0 views
          </small>
        </div>
        <div
          className="css-17xejub"
        />
        <div
          className="css-0"
        >
          <p
            className="chakra-text css-1aayjs6"
          >
            test
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
